http-server/src/server.rs

```diff
@@ -521,13 +513,10 @@ impl<L: Logger> ServiceData<L> {
 					None => return Ok(malformed()),
 				};
 
-				let allowed_headers = acl.allowed_headers().to_cors_header_value();
-				let allowed_header_bytes = allowed_headers.as_bytes();
-
 				let res = hyper::Response::builder()
 					.header("access-control-allow-origin", origin)
 					.header("access-control-allow-methods", "POST")
-					.header("access-control-allow-headers", allowed_header_bytes)
+					.header("access-control-allow-headers", "*".as_bytes())
```

Comment:
hmm, this is now handled by the tower middleware and we don't know which ones are whitelisted.

so `*` might be wrong?

Reply:
Hm yeah, can't this entire OPTIONs handling go away? We don't care about it at all here any more if the CORS middleware takes care of it.

Reply:
(and remember to update the comment on the Method::POST match too, and presumable we don't need to return an "access-control-allow-origin" header either on that?)

Reply:
right I guess we could rid of https://github.com/paritytech/jsonrpsee/pull/851/files#diff-f82cd7c573676fe64905b975da3d3c8cdb9a251297b53da66aabc2d027cdeaccR532 too then

Reply:
Yep, have entirely removed those. The functionality works as before. To satisfy CORS, we still need to rely on upstream layers, but that's ok

Reply:
that much better, removed code is a win :)